app/src/pages/settings/RetentionPoliciesTable.tsx

```diff
@@ -0,0 +1,178 @@
+import { RetentionPoliciesTablePoliciesQuery } from "./__generated__/RetentionPoliciesTablePoliciesQuery.graphql";
+export const RetentionPoliciesTable = ({
+  query,
+}: {
+  query: RetentionPoliciesTable_policies$key;
+}) => {
+  const [data] = useRefetchableFragment<
+    RetentionPoliciesTablePoliciesQuery,
+    RetentionPoliciesTable_policies$key
+  >(
+    graphql`
+      fragment RetentionPoliciesTable_policies on Query
+      @refetchable(queryName: "RetentionPoliciesTablePoliciesQuery") {
+        projectTraceRetentionPolicies {
+          edges {
+            node {
+              id
+              name
+              cronExpression
+              rule {
+                __typename
+                ... on TraceRetentionRuleMaxCount {
+                  maxCount
+                }
+                ... on TraceRetentionRuleMaxDays {
+                  maxDays
+                }
+                ... on TraceRetentionRuleMaxDaysOrCount {
+                  maxDays
+                  maxCount
+                }
+              }
+              projects {
+                edges {
+                  node {
+                    name
+                    id
+                  }
+                }
+              }
+            }
+          }
+        }
+      }
+    `,
+    query
+  );
+  const tableData = data.projectTraceRetentionPolicies.edges.map(
+    (edge) => edge.node
+  );
+
+  const columns: ColumnDef<(typeof tableData)[number]>[] = [
+    {
+      header: "Name",
+      accessorKey: "name",
+    },
+    {
+      header: "Schedule",
+      accessorKey: "cronExpression",
+      cell: ({ row }) => {
+        return cronstrue.toString(row.original.cronExpression);
+      },
+    },
+    {
+      header: "Rule",
+      accessorKey: "rule",
+      cell: ({ row }) => {
+        const rule = row.original.rule;
+        if (rule.__typename === "TraceRetentionRuleMaxCount") {
+          return `${rule.maxCount} traces`;
+        }
+        if (rule.__typename === "TraceRetentionRuleMaxDays") {
+          if (rule.maxDays === 0) {
+            return "Infinite";
+          }
+          return `${rule.maxDays} days`;
+        }
+        if (rule.__typename === "TraceRetentionRuleMaxDaysOrCount") {
+          return `${rule.maxDays} days or ${rule.maxCount} traces`;
+        }
+        if (rule.__typename === "%other") {
+          return "Unknown";
+        }
+        assertUnreachable(rule);
+      },
+    },
+    {
+      header: "Projects",
+      accessorKey: "projects",
+      cell: ({ row }) => {
+        return (
+          <ul
+            css={css`
+              li {
+                display: inline;
+              }
+              li:not(:first-child):before {
+                content: ", ";
+              }
+            `}
+          >
+            {row.original.projects.edges.map((edge) => (
+              <li key={edge.node.id}>
+                <Link to={`/projects/${edge.node.id}/config`}>
```

Comment:
Should this be restricted to admins?

Reply:
I was gonna make editing restricted. No harm to know I was thinking. We have to think about the un-authenticated version a tad too.